templates/virt-install-script.sh.j2

```diff
@@ -1,29 +1,30 @@
-#!/bin/bash
-
-if [ ! -f /etc/libvirt/qemu/{{ inventory_hostname }}.xml ]; then
-
 virt-install \
   --name={{ inventory_hostname }} \
-  --os-variant=rhel6 \
-{% if virt_install_cpu_model %}
-  --cpu={{ virt_install_cpu_model }} \
-{% else %}
-  --cpu=host \
-{% endif %}
   --vcpus={{ vcpus }} \
   --ram={{ ram }} \
 {% for disk in disks %}
-  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format=raw \
+  --disk path={{ disk.path }},cache=writethrough,device=disk,bus=virtio,size={{ disk.size }},format={{ disk.format | default('raw') }} \
 {% endfor %}
 {% for bridge in bridges %}
   --network bridge={{ bridge }},model=virtio \
 {% endfor %}
   --connect=qemu:///system \
-  --location={{ install_url }} \
-  --graphics=vnc,keymap="fi" \
-  --noautoconsole \
   --autostart \
+  --noautoconsole \
+  --events on_poweroff=preserve \
```

Comment:
You could move the `on_poweroff=preserve` setting to be only applied when doing image based deployments as some other virt-install arguments are already applied.